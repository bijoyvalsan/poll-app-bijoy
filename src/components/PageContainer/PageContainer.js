import React from "react";
import PropTypes from "prop-types";

import {
  StyledContainer,
  StyledTitle,
  StyledHeaderBox
} from "./PageContainer.styled";

const PageContainer = ({ title, children, renderActions }) => (
  <StyledContainer>
    <StyledHeaderBox>
      {title ? (
        <StyledTitle data-testid="page-container-title">{title}</StyledTitle>
      ) : null}
      {renderActions ? renderActions() : null}
    </StyledHeaderBox>
    {children}
  </StyledContainer>
);

PageContainer.defaultProps = {
  title: undefined,
  renderActions: undefined
};

PageContainer.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]),
  title: PropTypes.string,
  renderActions: PropTypes.func
};

export default PageContainer;
