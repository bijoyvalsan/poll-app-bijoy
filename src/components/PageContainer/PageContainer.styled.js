import styled from "styled-components";

export const StyledContainer = styled.div`
  display: flex;
  flex-direction: column;
  box-sizing: border-box;
  margin-bottom: 24px;
`;

export const StyledHeaderBox = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const StyledTitle = styled.h2`
  color: #424242;
  ont-size: 24px;
`;
