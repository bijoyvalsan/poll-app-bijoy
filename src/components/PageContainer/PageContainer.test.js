import React from "react";
import { render } from "@testing-library/react";

import PageContainer from "./PageContainer";

describe("PageContainer Component", () => {
  it("should render with default PageContainer styles", () => {
    const { asFragment } = render(
      <PageContainer title="test123">
        <div>123</div>
      </PageContainer>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should contain title test123", () => {
    const { queryByTestId } = render(
      <PageContainer title="test123">
        <div>123</div>
      </PageContainer>
    );
    const titleElement = queryByTestId("page-container-title");
    expect(titleElement.textContent).toBe("test123");
  });
});
