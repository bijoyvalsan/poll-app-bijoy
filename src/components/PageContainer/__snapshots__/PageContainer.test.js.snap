// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContainer Component should render with default PageContainer styles 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnxRM bgCCbn"
  >
    <div
      class="sc-gtsrHT ebpWAq"
    >
      <h2
        class="sc-dlnjwi ccgZlH"
        data-testid="page-container-title"
      >
        test123
      </h2>
    </div>
    <div>
      123
    </div>
  </div>
</DocumentFragment>
`;
